src/main/java/redis/clients/jedis/util/GuavaCSC.java

```diff
@@ -0,0 +1,92 @@
+package redis.clients.jedis.util;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hasher;
+import java.util.concurrent.TimeUnit;
+import redis.clients.jedis.ClientSideCache;
+import redis.clients.jedis.CommandObject;
+
+public class GuavaCSC extends ClientSideCache {
+
+  private static final int DEFAULT_MAXIMUM_SIZE = 10_000;
```

Comment:
I'm more convinced now that we need some layer between these two cache types, within the package. Note that they share the same finals,  but different functions. It sounds like we should set the general rules in a parent, then the override (is HashFunction here) in the child.

Or.. is this overkill for two classes?